### PR TITLE
WebUI: Fix overflow-menu and context-menu positioning for small viewports

### LINKED ISSUE #5299: WebUI: Hamburger menu is not scrollable if window is too small
### What is the issue?

grafik
The scrollbar on the right scrolls just a tiny bit. The top 1/3 of the menu contents is basically missing with no means to access it.
Seen in Firefox on Windows

### Which application of Transmission?

Web client

### Which version of Transmission?

4.0.2

### PR DESCRIPTION
**Pull request by Rukario**
Allows main menu accessible on smaller browser window.

Fixes #5299

Additional fix made for context menu.

### CODE REVIEW

**Comment by tearfur** [comment]
Is this meant as an alternative to #5820?

**Comment by ckerr** [comment]
They do seem to overlap.

@dareiff, @Rukario which of these should be merged?

**Review by ckerr — COMMENTED** [review]
Minor lint warning:

/home/runner/work/transmission/transmission/web/src/transmission.js
  31:3  error  'movePopup' is defined but never used  no-unused-vars

**Comment by dareiff** [comment]
> They do seem to overlap.
> 
> @dareiff, @Rukario which of these should be merged?

Yeah, they’re different solutions to the issue. I think it’s up to UI review!

IMHO, we shouldn’t be changing the padding on the overflow menu, nor putting it so close to the edge:
(#5827 on the left, #5820 on the right)

Screenshot 2023-07-28 at 8 37 55 AM
Screenshot 2023-07-28 at 8 37 28 AM

And in smaller viewports #5820 uses a height that doesn’t extend below the bottom edge in favor of scrolling the popup overflow menu:
Screenshot 2023-07-28 at 8 41 59 AM

(versus)

Screenshot 2023-07-28 at 8 43 35 AM

**Comment by GaryElshaw** [comment]
Derek is the most intimate with the web ui, and i approve of his work fwiw.

**Comment by ckerr** [comment]
- Biggest picture though is that when there are conflicting PRs I almost always prefer for the coders to see if they can work it out among themselves. Both of y'all have good PRs here so I want to make sure that whatever gets merged, both of you feel encouraged to do more PRs.

- From the UI screenshots, I'd be fine either of these. (But as @GaryElshaw already knows from other tickets, I'm probably not the most discerning UI reviewer :smile_cat:)

- From a code perspective, I like this PR's JS code simplifications in web/src/*js and would probably try to pick those up regardless of which PR gets used.

- In terms of loose ends, it's not clear to me if https://github.com/transmission/transmission/pull/5820#issuecomment-1652015188 has been addressed? I don't see secondary scrolls in any of these screenshots but agree those can be annoying.

tldr @dareiff + @Rukario if the two of you can find consensus either by picking PR A or B or by making a hybrid of the two, that's my preference

**Comment by Rukario** [comment]
Left: This PR, Middle: Original, Right: @dareiff PR
transmission screenshot

I've looked at padding thing because the original didn't have big gap on the right back then. I've updated to put back the padding changes, having updated codes again to replicate original using a new CSS property (margin: 6px) which also moved down a bit but looked better having so.

**Comment by Rukario** [comment]
Left: This PR, Middle: Original, Right: @dareiff PR
transmission screenshot

I've looked at padding thing because the original didn't have big gap on the right back then. I've updated to put back the padding changes, having updated codes again to replicate original using a new CSS property (margin: 6px) which also moved down a bit but looked better.

**Comment by GaryElshaw** [comment]
No shade on @Rukario, but i aesthetically prefer the shorter menu and the additional padding.

**Comment by Rukario** [comment]
> No shade on @Rukario, but i aesthetically prefer the shorter menu and the additional padding.

I've put back the changes regarding padding (also hello @dareiff about this change), this PR should be just about fixing the bug. I don't want the big gap nor secondary scrolls. I've taken compromises for most of the changes I did when attempting at fixing #5299.

It seems min-width properties was removed that caused the menu to look narrower, you tell me if that's a good change. And what of the menu being pulled down by a few pixels because of margin propeties that surrounded it by 6px?

**Comment by Rukario** [comment]
Not 100% what it means that TeamCity build has failed

Anyways, also removed
> -webkit-overflow-scrolling: touch;

Dunno if that'll help, but more importantly that appears to be non-standard CSS property so removed anyways.

**Comment by ckerr** [comment]
> * When there are conflicting PRs I almost always prefer for the coders to see if they can work it out among themselves. Both of you have good PRs here so I'm not worried about this fix so much as I am making sure that at the end of the day, both of you feel encouraged to do future PRs.
> * From the UI screenshots, I'd be fine either of these. (But as @GaryElshaw already knows from other tickets, I'm probably not the most discerning UI reviewer smile_cat)
> * From a code perspective, I like this PR's JS code simplifications in web/src/*js and would probably try to pick those up regardless of which PR gets used.
> * In terms of loose ends, it's not clear to me if [Add scrollability to overflow-menu and context-menu #5820 (comment)](https://github.com/transmission/transmission/pull/5820#issuecomment-1652015188) has been addressed? I don't see secondary scrolls in any of these screenshots but agree those can be annoying.
> 
> tldr @dareiff + @Rukario if the two of you can find consensus either by picking PR A or B or by making a hybrid of the two, that's my preference

@dareiff ping

**Comment by dareiff** [comment]
Sorry for the delay ya’ll, have some family visiting Seattle. If @Rukario has done testing on theirs, I’m A-OK with this PR going through.

**Comment by Rukario** [comment]
> Sorry for the delay ya’ll, have some family visiting Seattle. If @Rukario has done testing on theirs, I’m A-OK with this PR going through.

I tested across browsers and I have good confidence! I see that you did a lot of work on the web UI, it's improved a lot and it wouldn't have been done without you! Sorry if I'm getting in your way but there are a few more I can help make web UI better. I'm new to collaboration. I have been to HTML and several other computer languages (python, javascript, some batch files) for some time so I think I can get hang on it, I see room for improvements on Transmission side of the web UI. I use Transmission through command line (CLI). iPhone/iPad can do torrents through Transmission under iSH app (linux emulation) but it seems to crash upon serving web UI. I hope it can be usable the next time I want to access web UI when it works. On the other hand I dropped Tixati for Transmission for my desktop PCs because it's so clean and easy to use. It gets the job done without feature creeps.

**Comment by Rukario** [comment]
I have another PR awaiting for this to ship so there's no fork syncing conflicts. Is everything ok?

**Comment by ckerr** [comment]
> I have another PR awaiting for this to ship so there's no fork syncing conflicts. Is everything ok?

Everything's OK; I've just been super busy on non-Transmission work and I've been the bottleneck. Thanks for the other PRs!

**Comment by Rukario** [comment]
> > I have another PR awaiting for this to ship so there's no fork syncing conflicts. Is everything ok?
> 
> Everything's OK; I've just been super busy on non-Transmission work and I've been the bottleneck. Thanks for the other PRs!

Almost overlook your message but thanks, my anxiety is real and overwhelm is real, it can be hard to track things.

**Comment by ckerr** [comment]
Notes: Made the main menu accessible even on smaller displays.

**Comment by ckerr** [comment]
@Rukario I cherry-picked this back into `4.0.x` in 1458a7266a5aeba906e2ca26b45108bf25d341eb but the patch did require manual intervention so please feel free to double-check my work :eyes:

**Comment by Rukario** [comment]
@ckerr Wonderful! Having checked 4.0.x again with new changes, it looks like everything is in working order. You did fine!
